### ISSUE TITLE
Skins on Vanilla not is working!!

### ISSUE THREAD

**Issue by Jhair2015** [Barbosik/MultiOgar#365]
Please help skins not working on vanilla!!!

**Comment by SimonOrJ** [comment]
It looks like Agario devs figured out how to block skins.  It isn't working on my server as well.

**Comment by SimonOrJ** [comment]
Based on my findings, although the skin doesn't show only for the player using it, it's visible to other players.

In other words, if you name yourself `<fox>`, other players will see the fox skin on you.  You just can't see the fox skin on yourself.

**Comment by nuclearc** [comment]
haha you can change name in-game xD

**Comment by ItzLevvie** [comment]
@Barbosik You alive still?

**Comment by NatsuTheGreat** [comment]
I noticed this. It was weird.

**Comment by NatsuTheGreat** [comment]
But that's a dumb way of preventing it.

**Comment by Bull5072** [comment]
how to desactive vainilla mode

**Comment by SimonOrJ** [comment]
> how to desactive vainilla mode

You can use another client as listed here: https://github.com/Barbosik/MultiOgar#ogar-clients

**Comment by fayizan** [comment]
@Barbosik  please fix it

**Comment by nuclearc** [comment]
@fayizan it cant be fixed

**Comment by NatsuTheGreat** [comment]
@nuclearc What do you mean?

**Comment by fayizan** [comment]
@nuclearc why it cant be fixed..?

**Comment by nuclearc** [comment]
btw you can change skin using 'skin' command, idk about using skin from nickname.....

**Comment by NatsuTheGreat** [comment]
If others can see it, then it's probably some node updating that's delayed.

**Comment by fayizan** [comment]
@nuclearc can u please tell me the working way

**Comment by retslac** [comment]
@nuclearc your proposals will not work

I see no easy way of fixing this server-side.  Vanilla client doesn't use either name or skin anymore from the server updatepackets when the updated cell is your own cell.  Instead it directly uses the name and skin you chose in client.  You have no server-side control over your own name or skin anymore.  Even when you use console commands to change it, it will only change on every client but your own.  So i think you really need to change client for it to work again.

Maybe change server code so you would be playing with other player cells was my first thought (much like you would play with a minion).  But that wouldn't work because your own cells are needed on client for calculating mass, position, zoom amount, etc.

It might be possible however to change server code so it uses Spectate mode for playing.  In Spectate mode you do have server-side control over name, skin, position and zoom amount.  But this method would have some unwanted side effects like mass and stats not working anymore.  So this is not really a path i wanna take.  And not even sure if it works.

Anybody have any other ideas besides using another client?

**Comment by NatsuTheGreat** [comment]
Could it be something with packet 0?

**Comment by retslac** [comment]
Don't think so.  Nickname is received correctly in packet 0.  But you can only use it to update name and skin on other clients because your own client ignores it completely when sending back an updateNodes packet.

**Comment by NatsuTheGreat** [comment]
Can you go on discord? So we don't spam here.

**Comment by retslac** [comment]
ok im on

**Comment by NatsuTheGreat** [comment]
[Look what I found on Agar as spectating](http://prntscr.com/crkhu2)

**Comment by nuclearc** [comment]
agario changed something in protocol, i cant spawn anymore in offcial servers :(

**Comment by NatsuTheGreat** [comment]
Do you have any extensions running?

And lmao, sometimes agar.io even can't get the skin, yet it shows you that you have one in the store. And it was a really really big mistake making names client side.

**Comment by nuclearc** [comment]
no. i mean my custom client. i've worked on it like 4 months and researched almost all agario protocol, now Moneyclip changed protocol again and its not spawning :(

**Comment by NatsuTheGreat** [comment]
What does miniclip have to do with your client though? It's yours isn't?

**Comment by nuclearc** [comment]
My client is for bots

**Comment by NatsuTheGreat** [comment]
So it's like an add-on for agar?

**Comment by nuclearc** [comment]
no its like agario-client

**Comment by NatsuTheGreat** [comment]
[Well I think this could be your reason why](http://prntscr.com/crkuq3)

**Comment by cowpits** [comment]
the issue with skins is the latest agar.io client update as of october 2016. Miniclip is basically blocking skins. You can't see your skin but you can see other player's skins and they can see yours. You just won't see your own skin and on the leaderboard your name will show up like <skinname>yourname. 
It's very annoying and I'm getting a lot of complaints about it. Nothing anyone can do about it.

**Comment by nuclearc** [comment]
so actually client gets skin from shop....

**Comment by cowpits** [comment]
yes, the agar.io client will rely on agar.io shop as long as you play using http://agar.io/?ip=

**Comment by NatsuTheGreat** [comment]
@retslac Can you reply when you're here? I am getting at something with the name.

**Comment by nuclearc** [comment]
but i dont understand.... why they blocked skins in private servers?

**Comment by retslac** [comment]
@NatsuTheGreat  What you have in mind?  If you don't mind posting it here, please do.  I won't be around today but i'll read it later.

**Comment by cowpits** [comment]
skins on private servers are blocked if you use http://agar.io/?ip=  client. Basically the agar.io client. Miniclip is blocking skins because I think they don't want players to see the new skins for free. If you use skin in brackets you won't see it but other players will see the skin. I'm hoping they change that soon.

**Comment by NatsuTheGreat** [comment]
@retslac Hey really sorry for responding late, i'll try to get back to you ASAP.

**Comment by nuclearc** [comment]
what about this? :D

![image](https://cloud.githubusercontent.com/assets/8341270/19834765/875c18c4-9e86-11e6-943a-a57deac5e8aa.png)

**Comment by AlexandriteErikParacelsus** [comment]
@nuclearc noice

**Comment by cowpits** [comment]
@nuclearc where you spectating or actually playing?

**Comment by nuclearc** [comment]
i'm playing look at score and leaderboard

**Comment by cowpits** [comment]
Nice, how is it working for you? I know when I spectate I see all skins.

**Comment by nuclearc** [comment]
its v72

**Comment by cowpits** [comment]
v72? MultiOgar? 
This is what I see [https://goo.gl/photos/zNMtFE1ZVxvEoKpq8](url)
And when I spectate [https://goo.gl/photos/A6tNHEAzhoT3f1A9A](url)

**Comment by nuclearc** [comment]
i mean its Agar.io client v72. Search 'agar expose' and you'll find it

**Comment by cowpits** [comment]
OK but it's not the official Agar.io client that everyone uses, right? basically an extension. I opened a case with miniclip to see if they can ease off the skins issue.

**Comment by nuclearc** [comment]
doesnt matter :P v72 with skins is much better that agario

**Comment by Megabyte918** [comment]
@cowpits how do you know that while spectating you can see all skins? isnt it like that normally, but you can see everyone elses but your own skin?

**Comment by cowpits** [comment]
@Megabyte918 yeah that's the frustrating part is you can see skins when you spectate and even when playing you see every one else's skin but they see is the skin name in tag and their name, like <muu>cowpits and look how it shows on the leaderboard for the player. But everyone else see's the skin and the correct name in the leaderboard. It's really annoying what miniclip is done. I opened a request with them, I'm sure they'll end up blocking skins fully.

**Comment by cowpits** [comment]
I tell you what play on one of my servers and you'll see what I mean. Open two tabs one play on and the other spectate. 
Busy server: agar.io/?ip=gkclan.servegame.com:8888
New server: agar.io/?ip=gkclan.servegame.com:8880

**Comment by cowpits** [comment]
Try any skin you like with tags <sly>yourname when you play you won't see it, when you spectate it you'll see the sly skin.

**Comment by cowpits** [comment]
Here they are two tabs side by side lol
![image](https://cloud.githubusercontent.com/assets/22398975/19854800/5bb0bff6-9f47-11e6-88a1-82522c0d1837.png)

**Comment by cowpits** [comment]
![image](https://cloud.githubusercontent.com/assets/22398975/19854914/f8e47092-9f47-11e6-9d0b-1a80532b9569.png)

**Comment by cowpits** [comment]
Oh I forgot to mention that if you use a easter egg skin you'll be able to see it no problem.

**Comment by NatsuTheGreat** [comment]
Uhhh lie?

**Comment by fayizan** [comment]
I requested miniclip to bring back skins for private server ,they replied me this

http://imgur.com/4T5GkOq

**Comment by NatsuTheGreat** [comment]
I don't think they know what you're talking about.

**Comment by cowpits** [comment]
@fayizan i opened a case with them as well and explained in detail the issue and all i got was this. 

Hello Cowpits,

Thank you for contacting Miniclip Support.
We are not aware of any issue like the one you are describing. We have actually just tested it.

Make sure you are not using any unauthorized extension.
For any further question, please do not hesitate to contact us.

**Comment by fayizan** [comment]
Haha, I will mail them again....

**Comment by NatsuTheGreat** [comment]
Explain that you are talking about private servers if you haven't.

**Comment by cowpits** [comment]
And this is my reply to them: 

Cowpits, Nov 8, 14:56 GMT:
Hi Nuno,

Thank you for replying. Perhaps there's a misunderstanding of the issue. This is on Private servers using agar.io client. No extensions are being used. 
Basically on Private servers using agar.io client you put the skin name in tags like this <muu>yourname this gives the Moo skin. However since the October 2016 update, the skin doesn't show for the player but all other players can see the skin, except for the player himself. 
What I'm trying to say is that Miniclip is blocking skins on Private servers. Here's an example if you wish to test on my Private server http://agar.io/?ip=172.73.178.205:8880 if you spectate you see people's skins if you play you'll see skins but not yours. Pick a skin for example <sly> and you'll see what I mean. If you open two tabs, one to play and one to spectate you'll see the skin if you spectate yourself.

**Comment by Megabyte918** [comment]
Well that's helpful.

**Comment by fayizan** [comment]
@cowpits everyone knows that...😑
I m trying to make miniclip fix the skin problem completely but not by spectating own skin with multiboxing

**Comment by cowpits** [comment]
@fayizan I know that, but my gut feeling is they won't. What I posted is only one email back and forth. I tried to explain the problem to them but they don't care to really see it. If you're successful with getting through it would be awesome. I've almost given up lol.

**Comment by cowpits** [comment]
Oh looks like it removed the tags and the skin name from my email post. :(

**Comment by MastaCoder** [comment]
This is pretty pointless. Those skins are **paid** skins to use on Agar.io, pretty sure they don't want private servers using the skins and allowing people to use them for **free**.

Heck I think they did it on purpose to stop people from getting skins for free. You've seen Miniclip make wild business decisions without really helping players. Do you really think they would do this..?

**Comment by nuclearc** [comment]
maybe just make account system for private servers? like there can be some public api of agario account databases and MultiOgar can get user account info.
